### PR TITLE
TST: stats.quantile: add edge test case for axis=None && keepdims=True

### PR DESCRIPTION
**Pull request by zitongzhoueric**
Adds a test edge case for `stats.quantile` introduced in https://github.com/scipy/scipy/pull/22352, with arguments `axis=None, keepdims=True`

### CODE REVIEW

**Comment by zitongzhoueric** [comment]
This adds coverage to the following lines. With this addtion now `stats.quantile` is fully covered:
https://github.com/mdhaber/scipy/blob/25f943ec227134a8a9ef8e520b1a7de62724b731/scipy/stats/_quantile.py#L22-L25
https://github.com/mdhaber/scipy/blob/25f943ec227134a8a9ef8e520b1a7de62724b731/scipy/stats/_quantile.py#L249-L252

**Comment by mdhaber** [comment]
Would you add a test with scalar `p`, e.g. `0.5`?

File: scipy/stats/tests/test_quantile.py [hunk 1]
```diff
@@ -162,7 +162,9 @@ def test_integer_input_output_dtype(self, xp):
          ([[], []], 0.5, np.full(2, np.nan), {'axis': -1}),
          ([[], []], 0.5, np.zeros((0,)), {'axis': 0, 'keepdims': False}),
          ([[], []], 0.5, np.zeros((1, 0)), {'axis': 0, 'keepdims': True}),
-         ([], [0.5, 0.6], np.full(2, np.nan), {}),])
+         ([], [0.5, 0.6], np.full(2, np.nan), {}),
+         ([[1, 2], [3, 4]], [0.25, 0.5, 0.75], [[1.75, 2.5, 3.25]], 
+          {'axis': None, 'keepdims': True}),])
```
Comment (mdhaber) [inline scipy/stats/tests/test_quantile.py:167]: How about:
```suggestion
         ([], [0.5, 0.6], np.full(2, np.nan), {}),
         (np.arange(1, 28).reshape((3, 3, 3)), 0.5, [[[14.]]],
          {'axis': None, 'keepdims': True}),
         ([[1, 2], [3, 4]], [0.25, 0.5, 0.75], [[1.75, 2.5, 3.25]], 
          {'axis': None, 'keepdims': True}),])
```

**Review by mdhaber — APPROVED** [review]
If tests pass, this LGTM.
